src/d/actor/d_a_npc_photo.cpp

```diff
@@ -4,337 +4,2266 @@
+    mPhotoGalleryEventIdx = dComIfGp_evmng_getEventIdx("PHOTO_GALLERY",0xff);
+    mPhotoCounterTalk0EventIdx = dComIfGp_evmng_getEventIdx("PHOTO_COUNTER_TALK0",0xff);
+    mPhotoCounterTalk1EventIdx = dComIfGp_evmng_getEventIdx("PHOTO_COUNTER_TALK1",0xff);
+    mPhotoDateUB4EventIdx = dComIfGp_evmng_getEventIdx("PHOTO_DATE_UB4",0xff);
+    field_0x9A6 = mPhotoCounterTalk0EventIdx;
+
+    mEventCut.setActorInfo2(l_npc_staff_id[0], this);
+    
+    field_0x9B6 = 0;
+    field_0x9BC = false;
+    field_0x9BD = false;
+
+    fopAcM_SetMtx(this, mpMorf->getModel()->getBaseTRMtx());
+    attention_info.distances[fopAc_Attn_TYPE_TALK_e] = 173;
+    attention_info.distances[fopAc_Attn_TYPE_SPEAK_e] = 173;
+    attention_info.flags = fopAc_Attn_UNK1000000_e | fopAc_Attn_LOCKON_TALK_e | fopAc_Attn_ACTION_SPEAK_e;
+
+    eventInfo.setXyCheckCB(&daNpcPhoto_XyCheckCB);
+    eventInfo.setXyEventCB(&daNpcPhoto_XyEventCB);
+
+    if (field_0x9C1 == 4) {
+        field_0x9D7 = false;
+        field_0x9D8 = false;
+    } else {
+        field_0x9D7 = l_npc_dat[0].field_0x52;
+        field_0x9D8 = l_npc_dat[0].field_0x53;
+    }
+
+    field_0x9B2 = l_npc_dat[0].field_0x28;
+    mObjAcch.CrrPos(*dComIfG_Bgsp());
+
+    if(mObjAcch.GetGroundH() != C_BG_MIN_HEIGHT) {
+        current.pos.y = home.pos.y = mObjAcch.GetGroundH();
+    }
+
+    setMtx();
+    mpMorf->getModel()->calc();
+    mStts.Init(temp, 0xFF, this);
+    mCyl.Set(dNpc_cyl_src);
+    mCyl.SetStts(&mStts);
+    setCollision(&mCyl, current.pos + field_0x958, field_0x988, 150.0f);
+    field_0x6F8[0].Set(l_cyl_src2);
+    field_0x6F8[0].SetStts(&mStts);
+    field_0x6F8[1].Set(l_cyl_src2);
+    field_0x6F8[1].SetStts(&mStts);
+    return 4;
 }
 
 /* 00001048-000010A4       .text _delete__12daNpcPhoto_cFv */
 bool daNpcPhoto_c::_delete() {
-    /* Nonmatching */
+    dComIfG_resDelete(getPhaseP(), l_arcname_tbl[0]);
+    if(heap && mpMorf) {
+        mpMorf->stopZelAnime();
+    }
+    return true;
 }
 
 /* 000010A4-0000125C       .text _draw__12daNpcPhoto_cFv */
 bool daNpcPhoto_c::_draw() {
-    /* Nonmatching */
-}
+
+    J3DModel* model = mpMorf->getModel();
+    J3DModelData *model_data = model->getModelData();
+    g_env_light.settingTevStruct(TEV_TYPE_ACTOR, &current.pos, &tevStr);
+    g_env_light.setLightTevColorType(model, &tevStr);
+    mBtpAnm.entry(model_data, mFrame);
+    mpMorf->updateDL();
+    mBtpAnm.remove(model_data);
+
+    cXyz shadowPos(
+        current.pos.x,
+        current.pos.y + 150.0f,
+        current.pos.z
+    );
+
+    mShadowId = dComIfGd_setShadow(
+        mShadowId, 1, mpMorf->getModel(), &shadowPos, 800.0f, 20.0f,
+        current.pos.y, mObjAcch.GetGroundH(), mObjAcch.m_gnd, &tevStr
+    );
+
+    if (field_0x9C1 == 4) {
+        dSnap_Obj obj;
+
+        cXyz temp(
+            l_pso_photo->field_0x00,
+            l_pso_photo->field_0x04,
+            l_pso_photo->field_0x08
+        ); 
+        VECAdd(&temp, &current.pos, &temp);
```

Comment:
```suggestion
        temp += current.pos;
```
This also applies to the other instances of VECAdd in this TU